pgmpy/tests/test_models/test_BayesianNetwork.py

```diff
@@ -1142,14 +1162,38 @@ def test_predict(self):
                 "3",
             ]
         )
+        p3_complete = pd.concat(
+            [
+                self.titanic_data2[["Survived", "Sex"]][:30],
+                pd.DataFrame({"Pclass": p3_res}),
+            ],
+            axis=1,
+        )
 
-        np_test.assert_array_equal(p1_variable_elimination.values.ravel(), p1_res)
-        np_test.assert_array_equal(p1_belief_propagation.values.ravel(), p1_res)
+        self.assertTrue(
+            p1_variable_elimination.sort_index(axis=1).equals(
+                p1_complete.sort_index(axis=1)
+            )
+        )
+        self.assertTrue(
+            p1_belief_propagation.sort_index(axis=1).equals(
+                p1_complete.sort_index(axis=1)
+            )
+        )
 
-        np_test.assert_array_equal(p2_variable_elimination.values.ravel(), p2_res)
-        np_test.assert_array_equal(p2_approx_inference.values.ravel(), p2_res)
+        self.assertTrue(
+            p2_approx_inference.sort_index(axis=1).equals(
+                p2_complete.sort_index(axis=1)
+            )
+        )
+        self.assertTrue(
+            p2_variable_elimination.sort_index(axis=1).equals(
+                p2_complete.sort_index(axis=1)
+            )
+        )
 
-        np_test.assert_array_equal(p3.values.ravel(), p3_res)
+        # TODO: Tests for dataframes with NaN data?
```

Comment:
@Nimish-4 Do you still need to add some tests here?

Reply:
@ankurankan Kind of wanted your opinion if it is needed to test it currently. Like simply test whether the predicted NaNs are within an acceptable range (>60% correct or something like that). Beyond that, I think it is ready.

Reply:
@Nimish-4 For the `stochastic=False` case, we should get the exact same results everytime, so we can write tests for that checking that we get the correct dataframe output. For the `stochastic=True` case, I think we should add a `seed` argument to the `predict` method, and use that to get reproducible results that we can write tests for.

Reply:
@ankurankan The test case for `stochastic=False` is failing when the NaN mask is applied.